src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -362,17 +374,13 @@ private void UpdateWorkingDirectory()
                     return;
                 }
 
-                // check if we have already run init scripts for this solution, then simply return without executing init scripts
-                if (!string.IsNullOrEmpty(_currentSolutionDirectory) &&
-                    _currentSolutionDirectory.Equals(_solutionManager.SolutionDirectory, StringComparison.OrdinalIgnoreCase))
+                var latestRestore = _latestRestore;
+                var latestSolutionDirectory = _solutionManager.SolutionDirectory;
+                if (ShouldNoOpDueToRestore(latestRestore) &&
+                    ShouldNoOpDueToSolutionDirectory(latestSolutionDirectory))
```

Comment:
I'm still not sure of this no op logic. Wouldn't this logic now execute init.ps1 scripts everytime for a solution with non .net core projects? since in that case, `latestRestore` would always be null...